src/wasm/wasm-type.cpp

```diff
@@ -3172,16 +3164,90 @@ std::vector<HeapType> buildNominal(TypeBuilder& builder) {
   return heapTypes;
 }
 
+void replaceBasicHeapTypes(std::vector<std::unique_ptr<HeapTypeInfo>>& infos) {
+  for (auto& info : infos) {
+    struct BasicTypeReplacer : HeapTypeChildWalker<BasicTypeReplacer> {
+      void noteChild(HeapType* child) {
+        if (child->isBasic()) {
+          return;
+        }
+        auto* info = getHeapTypeInfo(*child);
+        if (info->kind == HeapTypeInfo::BasicKind) {
+          *child = info->basic;
+        }
```

Comment:
Reading this code, it checks if a child is basic. Then it checks if the info of the child is basic. Can you add a comment explaining when these two diverge, as that seems non-obvious?

Reply:
Done.